client/src/components/FormAction/FormAction.js

```diff
@@ -72,6 +86,17 @@ class FormAction extends SilverStripeComponent {
   }
 
   /**
+   * @return {boolean}
+   */
+  isConstructive() {
```

Comment:
I think `isPrimary()` or `isPrimaryType()` is a better name, it took me a while to work out what "constructive" meant and I doubt I'll remember a month later :)